WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,166 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using System.Threading;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+
+		public WebcamQrReader(Network network)
+		{
+			Network = network;
+		}
+
+		public void StartScanning()
+		{
+			if (ScanningTask is not { })
+			{
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
+					try
+					{
+						if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+						{
+							throw new NotImplementedException("This operating system is not supported.");
+						}
+						camera = OpenCamera();
+						RequestEnd = false;
+						Scan(camera);
+					}
+					catch (Exception exc)
+					{
+						Logger.LogError("QR scanning stopped. Reason:", exc);
+						ErrorOccured?.Invoke(this, exc);
+					}
+					finally
+					{
+						camera?.Release();
+					}
+				});
+			}
+		}
+
+		public async Task StopScanningAsync()
+		{
+			if (ScanningTask is { } task)
+			{
+				RequestEnd = true;
+				await task;
+
+				ScanningTask = null;
+			}
+		}
+
+		private void Scan(VideoCapture camera)
+		{
+			WriteableBitmap? lastBitmap = null;
+			WriteableBitmap? currentBitmap = null;
+			using QRCodeDetector qRCodeDetector = new();
+			while (!RequestEnd)
+			{
+				try
+				{
+					using Mat frame = new();
+					camera.Read(frame);
+					if (frame.Empty() || frame.Width == 0 || frame.Height == 0)
+					{
+						continue;
+					}
+					currentBitmap = ConvertMatToWriteableBitmap(frame);
+
+					NewImageArrived?.Invoke(this, currentBitmap);
+					lastBitmap?.Dispose();
+					lastBitmap = currentBitmap;
+
+					if (qRCodeDetector.Detect(frame, out Point2f[] points))
+					{
+						string qrCode = qRCodeDetector.Decode(frame, points, new Mat());
+						if (string.IsNullOrWhiteSpace(qrCode))
+						{
+							continue;
+						}
+						if (AddressStringParser.TryParse(qrCode, Network, out _))
+						{
+							BitcoinAddressFound?.Invoke(this, qrCode);
+							break;
+						}
+						else
+						{
+							InvalidAddressFound?.Invoke(this, qrCode);
+						}
+					}
+				}
+				catch (OpenCVException exc)
+				{
+					Logger.LogWarning(exc);
+					currentBitmap?.Dispose();
+				}
+			}
+			lastBitmap?.Dispose();
+			currentBitmap?.Dispose();
+		}
+
+		private VideoCapture OpenCamera()
+		{
+			VideoCapture camera = new();
+			if (!camera.Open(0))
```

Comment:
Maybe you can have a constant explaining what the `0` is about:

```suggestion
			if (!camera.Open(DefaultCameraId))
```